### PR TITLE
[MONIT-25884] Improve error message when buffer location has incorrec…

### PR DESCRIPTION
**Pull request by joannak-vmware**
…t permissions

### CODE REVIEW

File: proxy/src/main/java/com/wavefront/agent/queueing/TaskQueueFactoryImpl.java [hunk 1]
```diff
@@ -110,14 +111,31 @@ public Long value() {
     try {
       File lockFile = new File(lockFileName);
       if (lockFile.exists()) {
-        Preconditions.checkArgument(true, lockFile.delete());
+        if (!lockFile.delete()) {
```
Comment (laullon) [inline proxy/src/main/java/com/wavefront/agent/queueing/TaskQueueFactoryImpl.java:114]: Use `Files.Delete()` it give more info a better exceptions.
https://docs.oracle.com/javase/tutorial/essential/io/delete.html

File: proxy/src/main/java/com/wavefront/agent/queueing/TaskQueueFactoryImpl.java [hunk 2]
```diff
@@ -110,14 +111,31 @@ public Long value() {
...
+          "proxy");
+      return new TaskQueueStub<>();
+    } catch (OverlappingFileLockException e) {
+      logger.severe("Error requesting exclusive access to the buffer " +
+          "lock file " + lockFileName + " - please make sure that no other processes " +
+          "access this file and restart the proxy");
+      return new TaskQueueStub<>();
+    } catch (IOException e) {
+      logger.severe("Error requesting access to buffer lock file " + lockFileName + " Channel is " +
+          "closed or an I/O error has occurred - please restart the proxy");
```
Comment (laullon) [inline proxy/src/main/java/com/wavefront/agent/queueing/TaskQueueFactoryImpl.java:138]: Add the exception message to all log messages.

File: proxy/src/main/java/com/wavefront/agent/queueing/TaskQueueFactoryImpl.java (hunk 1 above)
Comment (laullon) [reply proxy/src/main/java/com/wavefront/agent/queueing/TaskQueueFactoryImpl.java:114]: or maybe best to use [deleteIfExists(Path)](https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#deleteIfExists-java.nio.file.Path-)

**Comment by joannak-vmware** [comment]
There are currently no unit tests for this, will be adding to this ticket

File: proxy/src/main/java/com/wavefront/agent/queueing/TaskQueueFactoryImpl.java [hunk 3]
```diff
@@ -110,15 +113,36 @@ public Long value() {
     try {
       File lockFile = new File(lockFileName);
       if (lockFile.exists()) {
-        Preconditions.checkArgument(true, lockFile.delete());
+        if (!Files.deleteIfExists(lockFile.toPath())) {
+          // no file with name found
+          throw new LogFilePathException();
```
Comment (laullon) [inline proxy/src/main/java/com/wavefront/agent/queueing/TaskQueueFactoryImpl.java:118]: We don't care if the file can't be deleted if not found, that means that there is no other proxy blocking the buffer, we only care if the file exists and can't be deleted, and that throw a exception. So can remove the `if`.

File: proxy/src/main/java/com/wavefront/agent/queueing/TaskQueueFactoryImpl.java [hunk 4]
```diff
@@ -110,15 +113,36 @@ public Long value() {
...
-    } catch (Exception e) {
-      logger.severe("WF-005: Error requesting exclusive access to the buffer " +
+      if (channel.tryLock() == null) {
+        throw new OverlappingFileLockException();
+      }
+    } catch (LogFilePathException e){
+          logger.severe("Error finding buffer lock file " + lockFileName +
+          " - please make sure file path is correct and restart the proxy: " + e);
+      return new TaskQueueStub<>();
+    } catch (DirectoryNotEmptyException e) {
```
Comment (joannak-vmware) [inline proxy/src/main/java/com/wavefront/agent/queueing/TaskQueueFactoryImpl.java:129]: Deleting catch as path will never be Directory (we create lockFile as new File above)

**Comment by joannak-vmware** [comment]
QA Testing:
Please test and verify the log messages are correct when:
1. When User running proxy does not have write permission
2. Have proxy running, and try to run a second proxy on same proxyid
3. Buffer path is set to path user does not have permission to write to

**Review by laullon — APPROVED** [review]

**Review by sbhakta-vmware — APPROVED** [review]
lgtm

**Pull request merged**